### PR TITLE
upgrade bn and add 2 curves

### PR DESCRIPTION
**Pull request by calvinmetcalf**

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
crypto-browserify/crypto-browserify#140

**Comment by calvinmetcalf** [comment]
removed p521 as that can be added back in a minor version

**Comment by calvinmetcalf** [comment]
@dcousens if we can get this merged then I can start work on the next phase upgrade of bn.js

File: package.json [hunk 1]
```diff
@@ -16,13 +16,13 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "bn.js": "^2.0.0",
-    "browserify-rsa": "^2.0.0",
+    "bn.js": "^3.0.0",
```
Comment (dcousens) [inline package.json]: why not `3.2.0`?  You have been switching between `3.2.0` and `3.0.0`.
Do we need `3.2.0`?

File: test/fixtures.json [hunk 2]
```diff
@@ -66,6 +66,13 @@
...
         "scheme": "DSA-SHA",
         "signature": "302c02147e0eb669879a758df5b3efae631fe51052852434021472e7d838db4356f9d5d305c78dbda65711259bfa"
+      },
+      {
+        "public": "LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUhZd0VBWUhLb1pJemowQ0FRWUZLNEVFQUNJRFlnQUVXVUIzL0RwOVZSWGR5VjhZY3lqRVFVWGRkMEFzUHh3bwpQTVpBRzJQbG1SVlhuRmMyZWhuT0tOVDVxdklGeXJTb0JYS285ckNmZi9CN0kwR3hpNzBNOHVML0JRS2Z5MWpBCkRvSEp1WndWVXpMa0RkZU1lL2ZiK1dXNFF1YUFVbHBDCi0tLS0tRU5EIFBVQkxJQyBLRVktLS0tLQo=",
+        "private": "LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1JR2tBZ0VCQkRBME5VbVMzOHoxcU1QK2R6Sjk4Y3J4bFkzZXhzN0tOVVlqZ25zODZzZ0ZBTmhPd3YyT09uVVgKcTcvbHZZUW1QN1NnQndZRks0RUVBQ0toWkFOaUFBUlpRSGY4T24xVkZkM0pYeGh6S01SQlJkMTNRQ3cvSENnOAp4a0FiWStXWkZWZWNWelo2R2M0bzFQbXE4Z1hLdEtnRmNxajJzSjkvOEhzalFiR0x2UXp5NHY4RkFwL0xXTUFPCmdjbTVuQlZUTXVRTjE0eDc5OXY1WmJoQzVvQlNXa0k9Ci0tLS0tRU5EIEVDIFBSSVZBVEUgS0VZLS0tLS0K",
+        "message": "ECDSA with P-384",
+        "scheme": "ecdsa-with-SHA1",
+        "signature": "3066023100fb65dfe219093dbb7b468ccded39931d845ff73a2386b9fd7a6a9860c5cdc56f120370dfa6b0441e14d30e5bfca371740231008f603c40136c059eb8f2de65fb9e773a79ed59102dcc1bfcc863c254f93a8139d579f952fa9e9512ba4ef75660b7565a"
```
Comment (dcousens) [inline test/fixtures.json]: Can we add descriptions to these fixtures?
Hard to tell why they are important.

File: curves.js [hunk 3]
```diff
@@ -6,3 +6,5 @@ exports['1.3.132.0.33'] = 'p224'
 exports['1.2.840.10045.3.1.1'] = 'p192'
 
 exports['1.2.840.10045.3.1.7'] = 'p256'
+
+exports['1.3.132.0.34'] = 'p384'
```
Comment (dcousens) [inline curves.js]: No `p521`?

**Comment by dcousens** [comment]
> removed p521 as that can be added back in a minor version

Why not just add both? Any reason?

**Comment by calvinmetcalf** [comment]
P521 needed a fix in elliptical that just landed but is part of the next
major version bump

**Comment by dcousens** [comment]
@calvinmetcalf should we just wait and land both?

File: test/fixtures.json (hunk 2 above)
Comment (dcousens) [inline test/fixtures.json]: I assume its just to test `p384`?

Comment (calvinmetcalf) [inline test/fixtures.json]: The message doubles as a description

File: package.json (hunk 1 above)
Comment (calvinmetcalf) [inline package.json]: Some I manually incremented the major digit and others I used npm

**Comment by calvinmetcalf** [comment]
We have to bump a bunch of stuff to the next version of bn.js before we can do this one.

**Comment by calvinmetcalf** [comment]
GREEN! thanks to 3158625f8980646b297a36577590949505db0917

**Comment by dcousens** [comment]
ACK :+1:

**Pull request merged**